### PR TITLE
Fix inconsistencies in IMetadataAssemblyImport

### LINKED ISSUE #26022: IMetaDataAssemblyImport::FindAssembliesByName description incorrect on summary page
The summary page for this interface says
> FindAssembliesByName Method | Gets an array of **mdAssemblyRef tokens** for the assemblies with the specified name.

but it does not return tokens; it returns IMetadataAssemblyImport interface pointers (as `IUnknown *`).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a3c1a82f-2418-22fb-3d0e-78c7f84a8fd1
* Version Independent ID: 8f8ba09d-e154-7315-dae5-1524cefe9fc4
* Content: [IMetaDataAssemblyImport Interface](https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/metadata/imetadataassemblyimport-interface)
* Content Source: [docs/framework/unmanaged-api/metadata/imetadataassemblyimport-interface.md](https://github.com/dotnet/docs/blob/main/docs/framework/unmanaged-api/metadata/imetadataassemblyimport-interface.md)
* Product: **dotnet-framework**
* GitHub Login: @gewarren
* Microsoft Alias: **dotnetcontent**

### LINKED ISSUE #26023: IMetaDataAssemblyImport::FindAssembliesByName in/out inconsistency
The *Syntax* section says
>  [out] IUnknown    *ppIUnk[],

but the *Parameters* section says
> ppIUnk
> [in]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7dbb70b8-752c-c8ad-8daa-94766af14952
* Version Independent ID: 2ab9b0f4-a002-56d5-efad-17599b256f0b
* Content: [IMetaDataAssemblyImport::FindAssembliesByName Method](https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/metadata/imetadataassemblyimport-findassembliesbyname-method)
* Content Source: [docs/framework/unmanaged-api/metadata/imetadataassemblyimport-findassembliesbyname-method.md](https://github.com/dotnet/docs/blob/main/docs/framework/unmanaged-api/metadata/imetadataassemblyimport-findassembliesbyname-method.md)
* Product: **dotnet-framework**
* GitHub Login: @gewarren
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #26022, fixes #26023.

[Source code](https://github.com/dotnet/runtime/blob/b0159122f3570decd8ced6228681a210e2711de6/src/coreclr/inc/cor.h#L1355-L1361).

### CODE REVIEW

**Comment by gewarren** [comment]
@tommcdon Would you be able to verify that these changes are correct, or point me to someone who could verify them?

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: when ready

**Review by tommcdon — CHANGES_REQUESTED** [review]
I believe the existing documentation for in/out parameters on FindAssembliesByName is correct: https://github.com/dotnet/runtime/blob/b0159122f3570decd8ced6228681a210e2711de6/src/coreclr/inc/cor.h#L1355-L1361

File: docs/framework/unmanaged-api/metadata/imetadataassemblyimport-interface.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ Provides methods to access and examine the contents of an assembly manifest.
 |[EnumExportedTypes Method](imetadataassemblyimport-enumexportedtypes-method.md)|Gets an interface pointer to an enumerator that contains the `mdExportedType` tokens of the COM types referenced by the assembly in the current metadata scope.|  
 |[EnumFiles Method](imetadataassemblyimport-enumfiles-method.md)|Gets an interface pointer to an enumerator that contains the `mdFile` tokens of the files referenced by the assembly in the current metadata scope.|  
 |[EnumManifestResources Method](imetadataassemblyimport-enummanifestresources-method.md)|Gets an interface pointer to an enumerator that contains the `mdManifestResource` tokens of the resources referenced by the assembly in the current metadata scope.|  
-|[FindAssembliesByName Method](imetadataassemblyimport-findassembliesbyname-method.md)|Gets an array of `mdAssemblyRef` tokens for the assemblies with the specified name.|  
+|[FindAssembliesByName Method](imetadataassemblyimport-findassembliesbyname-method.md)|Gets an array of `IMetaDataImport` interface pointers for the assemblies with the specified name.|
```
Comment (gewarren) [inline docs/framework/unmanaged-api/metadata/imetadataassemblyimport-interface.md:32]:
```suggestion
|[FindAssembliesByName Method](imetadataassemblyimport-findassembliesbyname-method.md)|Gets an array of `IMetaDataAssemblyImport` interface pointers for the assemblies with the specified name.|  
```

**Review by tommcdon — APPROVED** [review]
Looks good to me - thank you!

**Comment by gewarren** [comment]
> Looks good to me - thank you!

Thanks for reviewing and providing that link!

**Pull request merged**